tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -349,6 +372,48 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
+        internal TraceTagCollection AddMissingPropagatedTags()
```

Comment:
nit: I'd probably make this `void` return type instead of returning `TraceTagCollection`. Was there a specific reason you chose to return it?

Reply:
Yeah, it's a little weird, but the caller is doing something with the tags, and this method already goes through the trouble of finding which collection to use (or initialize a new one), so I was trying to avoid repeating that logic in the caller.

But you're right. I'll look into moving this logic to the caller instead.

Reply:
This was only called `DatadogContextPropagator` (aside from tests), so I moved part of the logic there and the rest to `TraceTagCollection`, removing this method from `SpanContext`. Thanks!